### DISCUSSION TITLE
Testing a long running WebSocket

### DISCUSSION THREAD

**Discussion by ibraEssam** [fastapi/fastapi#11139] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, HTTPException
app = FastAPI()

@app.websocket("/process/dummy_ws")
async def websocket_dummy(websocket: WebSocket):
    await websocket.accept()
    data = await websocket.receive_text()
    while True:
        await websocket.send_text(data)
        await asyncio.sleep(0.1)
```

### Description

I'm trying to test the example simple Webscoket using the following pytest test.
 
```python
from app.main import app  
from fastapi.testclient import TestClient
import pytest

@pytest.fixture()
def client():
    yield TestClient(app)

def test_dummy_ws(client):
    with client.websocket_connect("/process/dummy_ws") as websocket:
        websocket.send_text("test")
        data = websocket.receive_text()
        assert data == "test"
```

The test hangs forever and never finishes. Is there something missing im not aware of? 

### Operating System

Linux

### FastAPI Version

0.103.2

### Pydantic Version

2.6.1

### Python Version

Python 3.8.10

**Comment by Kludex** [comment, marked as answer]
Bump to the latest FastAPI.

**Reply by ibraEssam** [reply]
oh that's worked. version `0.109.2` works fine. Thanks for the help.
